### DISCUSSION TITLE
How can I access a resource or file from one Homebrew formula in another's `install`?

### DISCUSSION THREAD

**Discussion by aryairani** [Homebrew/discussions#1636] [Tap maintenance and Homebrew development] [answered]
I have one formula "A" that installs stuff in its `share` directory, and I want to know how to read the directory from another formula("B")'s `install`.  Is it possible?

Do I need to install A's resources into `etc.` instead of `share`?  Or how can I share data between packages?

I would use B's data as a `resource`, but it's got its own build process, so I thought having its own formula would be better.

Thanks in advance!

**Comment by nandahkrishna** [comment, marked as answer]
You should be able to use `Formula["A"].share` to get the path of `A`'s `share` directory. Of course, `B` should depend on `A`.

Update: You can take a look at `sng` in `homebrew/core` for an example.

**Reply by aryairani** [reply]
Thanks — I had tried this but misunderstood that the string passed to `Formula` should be the kebab-case, not camel-case, and got a `FormulaUnavailableError`.  So `Formula["a"].share`, I suppose, rather than `Formula["A"].share`.

**Reply by aryairani** [reply]
Is it possible to create a symlink to whatever version over `Formula["a"].share` happens to be `brew link`ed?  Currently it picks the version that was available at the time that `B` was installed:  
```
lrwxr-xr-x  1 arya  admin    34 Jun  7 13:06 ui -> ../../ucm-codebase-ui/1.0.M2/share
```
(Feel free to let me know if I should open a separate discussion.)

**Reply by nandahkrishna** [reply]
You could use `opt_share`:
```
irb(main):001:0> Formula["wget"].opt_share
=> #<Pathname:/usr/local/opt/wget/share>
```
